enhancements/proposals/1978-scylladbdatacenter-api/README.md

```diff
@@ -0,0 +1,741 @@
+  To preserve backward compatibility controllers will support enabling it by setting special annotation 
+  on `v1alpha1.ScyllaDBDatacenter`. 
+  Controller migrating from `v1.ScyllaCluster` to `v1alpha1.ScyllaDBDatacenter` will set this annotation when required.
+
+### Helm chart
+
+Existing `scylla` helm chart is going to use existing `v1.ScyllaCluster`. It has several differences between what's in CRD
+and what's in the CRD, hence it cannot be simply mapped one-to-one.
+New helm chart called `scylladbdatacenter` will simply copy `spec` from `values` to a new CRD solving this issue making it 
+easily extensible, while keeping schema of CRD and helm schema in sync.
+
+### Scylla Manager integration
+
+Scylla Manager tasks do not fit into single datacenter, as Manager tasks work on the entire cluster. 
+These are removed from the spec and status of the new API until a better alternative for these is available.
+Existing controller reconciling these will continue using `v1.ScyllaCluster`, users wanting to use these, 
+won't be able to transition to new API until an extension is available.
+
+### Topology placement
+
+Currently, to provide a placement rules for rack, users have to provide a very lengthy `nodeAffinity`. 
+Even the simplest rules require providing 10'ish lines with 4'ish indentation levels. 
+This makes the entire resource huge and UX isn't great. 
+To improve it, new field `toplogyLabelSelector` added on datacenter and rack level, 
+provides a label selector which will be translated into placement nodeAffinity rules selecting nodes matching these labels. 
+Labels selectors from datacenter and rack level will be merged into final node affinity.
+
+To prevent misuse of this simplification keys of provided label selector will be validated against `topology.kubernetes.io/*` rule, 
+and usage of both `nodeAffinity` and `topologyLabelSelector` will be forbidden by the webhook.
+Users wanting to construct more complex placement rules will have to do it using `placement` spec.
+
+### Resource model
+
+Storage `capacity` is going to be merged into `resources` field under `scyllaDB`, but other settings like metadata
+and storage class name, will stay in place. 
+This makes it to be in sync on how PVC template is defined.
+Agent doesn't require storage, so only `cpu` and `memory` keys are required within its `resource` spec.
+
+```yaml
+scylladb:
+  resources:
+    requests:
+      cpu: 1
+      memory: 4Gi
+      storage: 100Gi
+scyllaDBManagerAgent:
+  resources:
+    requests:
+      cpu: 100m
+      memory: 100Mi
+
+```
+
+### Uniform datacenter
+
+The current flexibility of having non-uniform racks will be preserved.
+
+ScyllaDB recommendation is to have uniform racks.
+Current API required from users to provide resources on every rack.
+There are two characteristics that make racks uniform: number of nodes and allocated resources.
```

Comment:
It's described it consists of two properties, then these properties are discussed and after a separator there's an example of such API. What you would like to improve?

Reply:
> What you would like to improve?

It just doesn't really say anywhere that this proposal introduces a API for unified racks (other than the name of this section). It just says that the current API will be preserved and that ScyllaDB recommends to have unified racks and what makes racks uniform, so I was reading it waiting for the actual proposed changed but it was never there - unless I missed it.

So I'd say adding even something as simple as "ScyllaDB recommendation is to have uniform racks.. I propose to extend the API so that the racks can be defined uniformly in a simplified manner, while preserving the existing flexibility of having non-uniform racks by overriding dedicated specifications" would make this clearer.

Reply:
I see what you mean now. Added statement you suggested.